src/jdk.zipfs/share/classes/jdk/nio/zipfs/ZipFileSystem.java

```diff
@@ -3193,14 +3205,15 @@ private static boolean isZip64ExtBlockSizeValid(int blockSize) {
             /*
              * As the fields must appear in order, the block size indicates which
              * fields to expect:
+             *  0 - May be written out by Ant and Apache Commons Compress Library
              *  8 - uncompressed size
              * 16 - uncompressed size, compressed size
              * 24 - uncompressed size, compressed sise, LOC Header offset
              * 28 - uncompressed size, compressed sise, LOC Header offset,
              * and Disk start number
              */
             return switch(blockSize) {
-                case 8, 16, 24, 28 -> true;
+                case 0, 8, 16, 24, 28 -> true;
```

Comment:
Don't need to handle the zero case here if you rearrange the code in `readExtra()` as suggested above.